src/index.ts

```diff
@@ -77,10 +77,14 @@ export class MaticPOSClient extends SDKClient {
     if (!txHash) {
       throw new Error(`txHash not provided`)
     }
-    if (options && !options.from) {
-      throw new Error(`options.from is missing`)
+    if (options && !options.from && !options.fastProof) {
```

Comment:
This check fails if fastProof is false. And there is no need to check fastProof, existing dapps will not pass this option and it will fail for them. if it is not present then assume false.